virttest/vt_agent/core/data_dir.py

```
@@ -0,0 +1,119 @@
+    :rtype: str
+    """
+    data_dir_path = get_data_dir()
+    # Ensure data_dir_path itself exists, as mkdtemp might not create it.
+    # This should be handled by __main__.py, but a check here adds robustness.
+    if not os.path.exists(data_dir_path):
+        try:
+            os.makedirs(data_dir_path)
+        except OSError as e:
+            LOG.error(
+                "Failed to create data directory %s for temp dir: %s", data_dir_path, e
+            )
```

Ruff (0.12.2)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)